apps/server/src/modules/saga/type/report.ts

```diff
@@ -0,0 +1,45 @@
+import { EntityId } from '@shared/domain/types';
+import { ModuleName } from './saga-step';
+
+export const enum StepOperationType {
+	DELETE = 'delete',
+	UPDATE = 'update',
+}
+
+export type StepOperationReport = {
+	operation: StepOperationType;
+	count: number;
+	refs: EntityId[];
+};
+
+export type StepReport = {
+	moduleName: ModuleName;
+	operations: StepOperationReport[];
+};
+
+// comes from '@modules/deletion' - StatusModel
+// TODO: check if we need this
+export const enum StepStatus {
```

Comment:
this is specific for user deletion saga only. Not sure this is the correct place for it
Also the name should reflect that.

Reply:
Agreed, has to be refactored. I'm actually not sure if this is necessary at all. I put it there for now to remove the dependencies to `DeletionModule`.